tutorials/commands/flux-job-cancel.rst

```diff
@@ -0,0 +1,114 @@
+.. _flux-job-cancel:
+.. _flux-job-cancelall:
```

Comment:
Just a high level note - Dan and I were trying to remember how to cancel the other day and the intuitive thing (that follows other tools) would be `flux job cancel --all`. Maybe we could deprecate `cancelall` or (so the intuitive one is available) just provide both?

Reply:
I didn't happen to write `cancelall`, although I suspect it was because `flux job cancel` takes jobids as input while `flux job cancelall` does not.  So separating them out made more sense than having to deal with mixing up the logic of both together.  

But it's a fair point, perhaps bring up the `--all` option for a discussion in `flux-core`.

Reply:
I think those commands were modeled after Unix/Linux `kill` vs `killall`. 

Considering we're still building interfaces, this command could be considered a "plumbing" command in the future, and so I wonder if a whole tutorial on `flux job cancel` is necessary. 
Feels like a "canceling jobs" howto might be better?

Reply:
I think for the time being, in that `flux job cancel` is a command (and one we badly needed yesterday!) it should be included here and we can move it / change our minds about it when the CLI is refactored.

With respect to that, I have some pretty strong opinions about cli design so I hope you invite me to the conversation, of course bringing a silver spike and garlic in case I get out of hand :vampire: